awx/main/scheduler/kubernetes.py

```diff
@@ -0,0 +1,74 @@
+import time
+
+from django.conf import settings
+from kubernetes.client import Configuration as kubeconfig
+from kubernetes.client import CoreV1Api as v1_kube_api
+from kubernetes.client import ApiClient as kube_api_client
+
+from awx.main.utils.common import parse_yaml_or_json
+
+
+class PodManager(object):
+    credential = None
+    kube_api = None
+
+    def __init__(self, task):
+        self.task = task
+        self.credential = task.instance_group.credential
+        self.kube_api = self._kube_api_connection(self.credential)
+        self.pod_name = f"job-{task.id}"
+        self.pod_definition = self._pod_definition()
+
+    def deploy_pod(self):
+        if not self.credential.kubernetes:
+            raise RuntimeError
```

Comment:
A log or message for this runtime error might be helpful, e.g: `raise RuntimeError('Pod deployment cannot occur without a Kubernetes credential')`

Reply:
@shanemcd :wave: just wanted to double-check - did we want to add any sort of message here?